.github/workflows/build.yml

```diff
@@ -174,7 +174,7 @@ jobs:
         uses: actions/setup-java@v4
         with:
           distribution: corretto
-          java-version: 17
+          java-version: 21
 
       # Run Qodana inspections
       - name: Qodana - Code Inspection
```

Comment:
We should also update this build action to v2025.1.1 to keep in sync with the gradle plugin

Reply:
I just tried that, but it seems that anything greater than 2023.2 needs a quodana cloud account, and that requires admin privs on the github organization, which I don't have. So I've downgraded it to 2023.1